File: e-commerce/src/main/java/com/e_ticaret/e_commerce/service/CartService.java
package com.e_ticaret.e_commerce.service;

import com.e_ticaret.e_commerce.dto.CartDto;

public interface CartService {
    CartDto getCartByUserId(Long userId);
    void addToCart(Long userId, Long productId, int quantity);
    void removeFromCart(Long userId, Long productId);
    void clearCart(Long userId);
}

File: frontend-yeni/src/app/core/cart.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environment/environment';

@Injectable({
  providedIn: 'root'
})
export class CartService {
  private apiUrl = `${environment.apiUrl}/cart`;

  constructor(private http: HttpClient) {}

  getCart(userId: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/${userId}`);
  }

  addToCart(userId: number, productId: number, quantity: number): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/${userId}/add`, { productId, quantity });
  }

  removeFromCart(userId: number, productId: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${userId}/remove/${productId}`);
  }

  clearCart(userId: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${userId}/clear`);
  }
}
